src/app/dashboard/Missions/CreateMission.jsx

```diff
@@ -0,0 +1,480 @@
+  },
+  sideBySideSelect: {
+    display: "flex",
+    margin: "10px",
+  },
+  numSelector: {
+    width: "100px",
+  },
+  selectCreateButton: {
+    textAlign: "left",
+    padding: "10px",
+    color: "#3739B5",
+    maxWidth: "50%",
+  },
+  inputLabel: {
+    fontSize: "20px",
+  },
+}));
+
+const CreateMission = () => {
+  const classes = useStyles();
+  const [missionType, setMissionType] = useState();
+  const [associatedItems, setAssociatedItems] = useState([]);
+  // master mission object
+  const [newMissionObject, setNewMissionObject] = useState({});
+
+  // master submit
+  function submitMission(e) {
+    e.preventDefault();
+    setNewMissionObject({
+      missionId: uuidv4(),
+      createdAt: new Date(),
+      createdBy: "organizer", // make dynamic
+      missionType: missionType,
+      associatedItems: associatedItems,
+      fund: "",
+      recipient: {
+        name: "",
+        phoneNumber: "",
+      },
+      volunteer: {
+        name: "",
+        phoneNumber: "",
+      },
+      dropOff: {
+        address: "",
+        date: "",
+      },
+      pickUp: {
+        address: "",
+        date: "",
+      },
+      comment: "",
+    });
+    console.log(newMissionObject);
+  }
+
+  const handleChangeMissionType = (event) => {
+    setMissionType(event.target.value);
+  };
+
+  // This handles all the food box inputs that are added to food box missions
+  const [foodBoxInputCount, setFoodBoxInputCount] = useState([]);
+  function addToFBCount() {
+    setFoodBoxInputCount([...foodBoxInputCount, foodBoxInputCount.length + 1]);
+  }
+
+  return (
+    <Container maxWidth="sm">
+      <Typography className={classes.addMissionContainer} component="div">
+        <div className={classes.title}>
+          <AnnouncementIcon className={classes.icon} />
+          <Typography className={classes.titleHeader}>Add New Mission</Typography>
+        </div>
+        <Divider variant="middle" />
+        <Typography className={classes.inputHeader}>Mission Type</Typography>
+        <FormControl className={classes.form}>
+          <InputLabel
+            variant="outlined"
+            id="missionType"
+            shrink="true"
+            className={classes.inputLabel}
+          >
+            Select Mission Type
+          </InputLabel>
+          <Select
+            labelId="missionType"
+            variant="outlined"
+            className={classes.select}
+            value={missionType}
+            onChange={handleChangeMissionType}
+          >
+            <MenuItem value="food-box">Food Box</MenuItem>
+          </Select>
+        </FormControl>
+        {/* THIS PART SPAWNS BASED ON WHAT MISSION TYPE WAS SELECTED */}
+        {missionType === "food-box" ? (
+          <>
+            <Typography className={classes.inputHeader}>Select Food Box Options</Typography>
+            <FormControl className={classes.form}>
+              {/* DETERMINE FOOD BOX TYPES */}
+
+              {foodBoxInputCount.map((index) => {
```

Comment:
Have you tried tried making this map to an actual component instead of an anonymous function with state?
I don't know for sure but I'm willing to bet react cant reconcile states correctly like this, not to mention it's not very performant.